s32/drivers/s32k3/BaseNXP/include/Compiler.h

```diff
@@ -483,18 +483,27 @@ extern "C"{
 /**
  * @brief Compiler abstraction for allocating variables to nocache section
  */
-#ifdef CONFIG_NOCACHE_MEMORY
+#if defined(CONFIG_NOCACHE_MEMORY) || defined(CONFIG_NOCACHE_LOAD_MEMORY) 
 #ifdef __ZEPHYR__
 #include <zephyr/toolchain/gcc.h>
 #else
 #ifndef STRINGIFY
 #define STRINGIFY(x) #x
 #endif /* STRINGIFY */
 #endif /* __ZEPHYR__ */
+#endif /* CONFIG_NOCACHE_MEMORY || CONFIG_NOCACHE_LOAD_MEMORY */
+
+#if defined(CONFIG_NOCACHE_MEMORY)
 #define VAR_SEC_NOCACHE(name) __attribute__((section(".nocache." STRINGIFY(name))))
 #else
 #define VAR_SEC_NOCACHE(name)
-#endif /* CONFIG_NOCACHE_MEMORY */
+#endif
+
+#if defined(CONFIG_NOCACHE_LOAD_MEMORY)
+#define VAR_SEC_NOCACHE_LOAD(name) __attribute__((section(".nocache_load." STRINGIFY(name))))
```

Comment:
As I mentioned in the Zephyr PR, I do not think we should be creating an SOC specific Kconfig for loadable noncached memory. This feature is (at a minimum) useful for all Cortex-M7 SOCs

Reply:
When i revisited the bare-metal driver and had a discussion with the engineer from HAL team, he said that actually the variable is supposed and can be put in to a read only region. In the mean time, added a patch here: https://github.com/zephyrproject-rtos/hal_nxp/commit/25cc567e5c832413060a63e3c1bb5c76338f18c9